runtime/queries/metricsview.go

```diff
@@ -322,17 +334,29 @@ func convertToXLSXValue(pbvalue *structpb.Value) (interface{}, error) {
 
 func metricsViewDimensionToSafeColumn(mv *runtimev1.MetricsViewSpec, dimName string) (string, error) {
 	dimName = strings.ToLower(dimName)
+	dimension, err := findDimension(mv, dimName)
+	if err != nil {
+		return "", err
+	}
+	return safeName(getMetricsViewDimensionName(dimension)), nil
+}
+
+func findDimension(mv *runtimev1.MetricsViewSpec, dimName string) (*runtimev1.MetricsViewSpec_DimensionV2, error) {
 	for _, dimension := range mv.Dimensions {
 		if strings.EqualFold(dimension.Name, dimName) {
-			if dimension.Column != "" {
-				return safeName(dimension.Column), nil
-			}
-			// backwards compatibility for older projects that have not run reconcile on this dashboard
-			// in that case `column` will not be present
-			return safeName(dimension.Name), nil
+			return dimension, nil
 		}
 	}
-	return "", fmt.Errorf("dimension %s not found", dimName)
+	return nil, fmt.Errorf("dimension %s not found", dimName)
+}
+
+func getMetricsViewDimensionName(dimension *runtimev1.MetricsViewSpec_DimensionV2) string {
```

Comment:
This resolves the column name, not the dimension name – should something like `metricsViewDimensionColumn(dim)`